### ISSUE TITLE
InvalidArgumentError (see above for traceback): ConcatOp : Dimensions of inputs should match: shape[0] =

### ISSUE THREAD

**Issue by taewookim** [tensorflow/models#2600]
This is actively talked about on [SO](https://stackoverflow.com/questions/46108125/tensorflow-concatop-error-with-object-detection-api/46963131#46963131).

Setting batch_size to 1 seems to solve this problem.

Would like TF obj det API to comment: does this have any implication on model's accuracy? With batch_size=1, do we stop training model w/same criteria as being trained on batch_size > 1? (i.e. total loss < 1.0)

**Comment by skye** [comment]
Looks like @cipri-tom already posted on SO :)

**Comment by cipri-tom** [comment]
@taewookim it is difficult to say exactly when you should stop training. The loss is just a measure, it doesn't tell you much except that when it is going down, the model is learning. But there is no absolute threshold to tell you "DONE". Watch your validation measure e.g. `mAP@0.5` to see if your model is still improving.

**Comment by taewookim** [comment]
Thanks @cipri-tom 

I had some other [noob questions](https://stackoverflow.com/questions/47080259/tf-object-detection-api-random-sizes-for-each-class).. any deas?

**Comment by taewookim** [comment]
Hey @cipri-tom 

As per your request, i separated my [SO question](https://stackoverflow.com/questions/47080259/training-with-big-class-imbalance) with [this one](https://stackoverflow.com/questions/47098549/can-tf-object-detection-api-be-used-to-detect-objects-that-are-bounded-by-anothe), but jonathan from google answered that one.

have any idea where i'm going wrong with [this one](https://stackoverflow.com/questions/47129078/tf-object-detection-api-not-all-classes-are-being-detected-behaving-oddly)?
